dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyPath.java

```diff
@@ -25,52 +25,88 @@
 import java.util.List;
 import java.util.Objects;
 import java.util.stream.Collectors;
+import javax.annotation.Nullable;
 import org.eclipse.aether.artifact.Artifact;
 import org.eclipse.aether.graph.Dependency;
 
 /**
- * Path from the root to a node in a dependency tree, where a node is a dependency.
+ * Path from the root to a node in a dependency tree, where the root node is a Maven artifact and
+ * non-root nodes are dependencies.
+ *
+ * <p>The root node is null for the dependency trees generated for multiple artifacts by {@link
+ * DependencyGraphBuilder#buildFullDependencyGraph(List)}; otherwise the root node is not null.
  */
 public final class DependencyPath {
 
-  private List<Dependency> path = new ArrayList<>();
+  // The root of the dependency path. The project root is not a dependency.
+  private final Artifact root;
```

Comment:
The separation here is confusing and makes this class more complex. Why isn't the root just the first node in the path? At the very least the public API and documentation can present it as such, even if the internal implementation is more complex.

Reply:
Removed `getRoot()` API in favor of `get(0)`. Now the root and other artifacts are treated the same in `get(int)`.

> Why isn't the root just the first node in the path?

It's because the first node is an artifact (without a dependency). The other nodes are dependencies, carrying scope and optional flag.